src/modules/did/did.processor.ts

```diff
@@ -40,23 +40,41 @@ export class DIDProcessor {
 
   @Process(ADD_DID_DOC_QUEUE_NAME)
   public async processDIDDocumentAddition(job: Job<string>) {
-    this.logger.debug(`processing cache add for ${job.data}`);
-    const doc = await this.didService.addCachedDocument(job.data);
+    try {
+      this.logger.debug(`processing cache add for ${job.data}`);
+      const doc = await this.didService.addCachedDocument(job.data);
 
-    await this.pinClaims(doc.service.map((s) => s.serviceEndpoint));
+      await this.pinClaims(doc.service.map((s) => s.serviceEndpoint)).catch(
+        (err) => {
+          this.logger.error(`error pinning the claim: ${err}`);
```

Comment:
we need to decide if this error needs to be rethrown here

Reply:
This could be helpful.